linux/service.c

```diff
@@ -486,33 +486,33 @@ srv_main_loop(service_context sc)
     return 0;
 }
 
+
 /*
  * srv_create_context()
  *	create a service context
  */
+static service_context srvctx;
 service_context
 srv_create_context(void)
 {
-    service_context blah;
-
-    if ((blah = (service_context)malloc(sizeof(struct _servcxt))) == NULL) {
+    if ((srvctx = (service_context)malloc(sizeof(struct _servcxt))) == NULL) {
         sae_debug(SAE_DEBUG_ERR, "context was NULL\n");
         return NULL;
     }
-    blah->timer_id = 0;
-    FD_ZERO(&blah->readfds);
-    FD_ZERO(&blah->writefds);
-    FD_ZERO(&blah->exceptfds);
-    bzero((char *)blah->timers, (NTIMERS * sizeof(struct timer)));
-    bzero((char *)blah->inputs, (NFDS * sizeof(struct source)));
-    bzero((char *)blah->outputs, (NFDS * sizeof(struct source)));
-    blah->ntimers = blah->ninputs = blah->noutputs = 0;
-    blah->gbl_timer.tv_sec = 1000;
-    blah->gbl_timer.tv_nsec = 0;
-    blah->exceptor = NULL;
-    blah->keep_running = 1;
-
-    return blah;
+    srvctx->timer_id = 0;
+    FD_ZERO(&srvctx->readfds);
+    FD_ZERO(&srvctx->writefds);
+    FD_ZERO(&srvctx->exceptfds);
+    bzero((char *)srvctx->timers, (NTIMERS * sizeof(struct timer)));
```

Comment:
bzero takes a void *, no idea why this cast is here.

(I checked the manpage because "char *" means a string and it would be really weird if a buffer-zeroing primitive operates on strings instead of at least "unsigned char *".)